redfish-core/lib/systems.hpp

```diff
@@ -1741,34 +1741,78 @@ inline void getPowerMode(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
                 messages::internalError(aResp->res);
                 return;
             }
-            // Valid Power Mode object found, now read the current value
+
+            // Valid Power Mode object found, now read the mode variables.
             crow::connections::systemBus->async_method_call(
                 [aResp](const boost::system::error_code ec,
-                        const std::variant<std::string>& pmode) {
+                        PropertiesType& properties) {
                     if (ec)
                     {
                         // Service not available, no error, return no data
                         BMCWEB_LOG_DEBUG
-                            << "Service not available on PowerMode Get: " << ec;
+                            << "Service not available on PowerMode GetAll: "
+                            << ec;
                         return;
                     }
 
-                    const std::string* s = std::get_if<std::string>(&pmode);
-                    if (s == nullptr)
+                    BMCWEB_LOG_DEBUG << "Size: " << properties.size();
+
+                    for (const auto& property : properties)
                     {
-                        BMCWEB_LOG_DEBUG << "Unable to get PowerMode value";
-                        messages::internalError(aResp->res);
-                        return;
-                    }
+                        if (property.first == "PowerMode")
+                        {
+                            const std::string* s =
```

Comment:
Would prefer a variable more than a single character, oh well, it was there before. Upstream we have clang-tidy check for this